### DISCUSSION TITLE
question for Feite

### DISCUSSION THREAD

**Discussion by traveller1011** [hootnot/oanda-api-v20#198] [Q&A] [answered]
hootnot, 

question about the _orders_ in ```import oandapyV20.endpoints.orders as orders```

#### Context

i seem to NOT get a complete list of orders back. and i cannot figure out why.

```
params = {}
r = OrderList(accountID, params=params)
rv = CLIENT.request(r)
print(rv)
```
... i get a healthy, but empty, response:

```{'orders': [], 'lastTransactionID': '89'}```

the Oanda endpoint in question is here:

https://developer.oanda.com/rest-live-v20/order-ep/

![image](https://user-images.githubusercontent.com/93844523/210904760-4c2c6b2f-9095-4857-b747-8babecc7dfc6.png)

#### How I have been getting Orders in the interim

So I queried for Transactions list, which worked great. Then, from transactions, i deduced a list of order IDs that "must" exist. And queried for order details by individual pk.. You can see some orders here:

![image](https://user-images.githubusercontent.com/93844523/210905397-e8feb37e-828e-41c1-8f50-3892838da102.png)

... But i want to query a _list_ of my orders directly, instead of approaching it from the backdoor. I think I am just missing something dumb.

#### Question/Ask
Do you see where I might have gone wrong? If you just give me an inkling of idea, it would be appreciated. Also - i really love this package. And the style of code. Thank you for publishing this.

**Comment by hootnot** [comment, marked as answer]
Make sure you enter a limit order. Market orders will get executed directly and therefore not show up in the orderlist since that one does PENDING by default.

So, in your case, without specifying params, you only will see PENDING orders, and there will be none if you only issue MARKET orders.

I've just entered a limit sell EUR_JPY @145 and it gives me:

```
{
  "orders": [
    {
      "id": "387093",
      "createTime": "2023-01-06T08:28:04.607826616Z",
      "type": "LIMIT",
      "instrument": "EUR_JPY",
      "units": "-500000",
      "timeInForce": "GTD",
      "price": "145.000",
      "gtdTime": "2023-01-13T08:27:48.000000000Z",
      "triggerCondition": "DEFAULT",
      "partialFill": "DEFAULT_FILL",
      "positionFill": "DEFAULT",
      "state": "PENDING"
    }
  ],
  "lastTransactionID": "387093"
}

```

In your case you want your filled market orders and need to pass:

```
params = {"state": "FILLED"}
```

Or use the definitions: ```oandapyV20.definitions.orders```

```
import oandapyV20.definitions.orders as od
...
params = {"state": od.OrderState.FILLED}
```

... and then your (filled) MARKET orders will show up.

**Reply by traveller1011** [reply]
the orders in question are indeed market (FILLED) orders. 

this line of code nailed it:

```
params = {"state": "FILLED"}
```

....

```
{
    "orders": [
        {
            "id": "14",
            "createTime": "2022-09-09T03:18:12.125028186Z",
            "type": "MARKET",
            "instrument": "EUR_GBP",
            "units": "1",
            "timeInForce": "FOK",
            "positionFill": "DEFAULT",
            "state": "FILLED",
            "fillingTransactionID": "15",
            "filledTime": "2022-09-09T03:18:12.125028186Z",
            "tradeOpenedID": "15"
        },
        {
            "id": "12",
            "createTime": "2022-09-09T02:59:46.270557139Z",
            "type": "MARKET",
            "instrument": "GBP_JPY",
            "units": "-1",
            "timeInForce": "FOK",
            "positionFill": "REDUCE_ONLY",
            "state": "FILLED",
            "fillingTransactionID": "13",
            "filledTime": "2022-09-09T02:59:46.270557139Z",
            "tradeClosedIDs": [
                "11"
            ]
        },
        {
            "id": "10",
            "createTime": "2022-09-09T02:27:01.313655439Z",
            "type": "MARKET",
            "instrument": "GBP_JPY",
            "units": "1",
            "timeInForce": "FOK",
            "positionFill": "DEFAULT",
            "state": "FILLED",
            "fillingTransactionID": "11",
            "filledTime": "2022-09-09T02:27:01.313655439Z",
            "tradeOpenedID": "11"
        },
        {
            "id": "7",
            "createTime": "2022-09-08T03:01:36.264972732Z",
            "type": "MARKET",
            "instrument": "AUD_JPY",
            "units": "-1",
            "timeInForce": "FOK",
            "positionFill": "DEFAULT",
            "state": "FILLED",
            "fillingTransactionID": "8",
            "filledTime": "2022-09-08T03:01:36.264972732Z",
            "tradeOpenedID": "8"
        },
        {
            "id": "5",
            "createTime": "2022-09-08T02:55:30.498113064Z",
            "type": "MARKET",
            "instrument": "AUD_JPY",
            "units": "-1",
            "timeInForce": "FOK",
            "positionFill": "DEFAULT",
            "state": "FILLED",
            "fillingTransactionID": "6",
            "filledTime": "2022-09-08T02:55:30.498113064Z",
            "tradeOpenedID": "6"
        }
    ],
    "lastTransactionID": "135"
}
```

thank you very much for getting me back on track again!
